packer/core.go

```diff
@@ -397,12 +397,17 @@ func (c *Core) init() error {
 		allVariables[k] = v
 	}
 
+	// Regex to exclude any build function variable or template variable
+	// from interpolating earlier
+	// E.g.: {{ build `Host` }} or {{ .HTTPIP }}  won't interpolate now
+	renderFilter := "{{(\\s|)(build\\s\x60(.*?)\x60|\\.(.*?))(\\s|)}}"
```

Comment:
Sorry, I don't have all the context around the early interpolation. But would it be possible to just specify what variables you want to interpolate as opposed to the negation? Thinking of the case were variables change some how we may need to change this regex to further remove.

Reply:
I thought about that but I'm only sure about the ones that should not interpolate now. The other ones seem to be ok to interpolate, and that's a bigger list.

ps: I'm pushing a change to this one because actually it's just the template variables like `{{ .HTTPIP }}` the build ones will always fail before on validation.

Reply:
Got it. In that case the negation is a good choice :+1: